### PR TITLE
Packaged versions run as non-priveleged "leofs" user

### PR DESCRIPTION
**Pull request by vstax**
Details in https://github.com/leo-project/leofs/issues/638
This change requires migration step for existing users, which is yet to be documented.
Also, it requires changes to launch.sh script; without them, resulting package won't work correctly out of the box.

### CODE REVIEW

**Comment by mocchira** [comment]
@vstax 

LGTM.
however a little bit to be considered is remained here: https://github.com/leo-project/leofs/issues/638#issuecomment-290605596

also let me confirm that
Can I squash those commits into one when merging for making the commit history more clear/easy to manage.
Another choice instead of squashing into one is to make only two commits

- Enable to run as leofs user for debian
- Enable to run as leofs user for rpm

by squashing other commits for tiny bug fixes into those two.

**Comment by vstax** [comment]
@mocchira Yes, of course you can squash them. I try to split commits because it helps me to review during these splits and after that, but for final merge the clean history is best. I'm new to github but it looks like it allows to squash effortlessly on merge (without manual merge or rebase), so please do that.

I'll make that change that checks for version according to your suggestion, but please confirm if you really want to try to push this before 1.3.3 release or it's going into version after that so I can put the correct check.

File: deb/check_version.sh [hunk 1]
```diff
@@ -0,0 +1,61 @@
+#!/bin/sh
```
Comment (mocchira) [inline deb/check_version.sh]: dedupe {deb,rpm}/check_version.sh into common/check_version.sh and have a symlink on each with the relative path style.

File: deb/check_version.sh [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+go_on_after_warning () {
+# Don't warn twice
+	if [ "$continue_anyway" = true ]
+	then
+		return
+	fi
+
+	echo "Do you want to continue? The resulting package is not likely to work correctly."
+        echo "It is recommended that you use older version of packaging scripts instead."
```
Comment (mocchira) [inline deb/check_version.sh]: trim one space in front of `echo`

**Comment by mocchira** [comment]
@vstax LGTM. thanks for your contribution.

Note:
Before merging this PR and also the corresponding PR on leofs, I will broadcast this change on ML and other communication channels we have.

**Pull request merged**